### PR TITLE
s/grpc/http/g

### PR DESCRIPTION
**Pull request by balasanjay**

### CODE REVIEW

File: backend.go [hunk 1]
```diff
@@ -58,206 +50,54 @@ func runBe(configPath string) {
...
-	return &pb.FileInfo{
-		Name:             finfo.Name(),
-		SizeBytes:        finfo.Size(),
-		Mode:             uint32(finfo.Mode()),
-		LastModTimeNanos: finfo.ModTime().UnixNano(),
-	}
-}
+	fs := http.FileServer(http.Dir(config.ServePath))
+	mux.HandleFunc("/", func(rw http.ResponseWriter, req *http.Request) {
+		glog.Infof("Got request from user: %v", req.Header.Get("User"))
```
Comment (jonathanwei) [inline backend.go]: We probably don't need this at regular verbosity.

File: crypto.go [hunk 2]
```diff
@@ -38,27 +38,27 @@ func MustNewAEAD(key []byte) cipher.AEAD {
...
 func EncryptProto(aead cipher.AEAD, msg proto.Message, additionalData []byte) string {
-	plaintext := MustMarshalProto(msg)
+	plaintext := MarshalProtoOrPanic(msg)
 
 	nonce := getNonce(aead.NonceSize())
 
 	// Encrypt in-place.
 	ciphertext := plaintext
-	ciphertext = aead.Seal(ciphertext, nonce, plaintext, additionalData)
+	ciphertext = aead.Seal(ciphertext[:0], nonce, plaintext, additionalData)
```
Comment (jonathanwei) [inline crypto.go:47]: Heh.

File: frontend_http.go [hunk 3]
```diff
@@ -15,9 +18,36 @@ import (
 	pb "github.com/jonathanwei/fproxy/proto"
 )
 
-func runHttpServer(config *pb.FrontendConfig, client pb.BackendClient) {
+func runHttpServer(config *pb.FrontendConfig) {
```
Comment (jonathanwei) [inline frontend_http.go:21]: We can refactor this with `runBe` later.

File: frontend_http.go [hunk 4]
```diff
@@ -31,11 +61,34 @@ func runHttpServer(config *pb.FrontendConfig, client pb.BackendClient) {
...
 	}
 
+	backendURL, err := url.Parse(config.GetBackend().Addr)
+	if err != nil {
+		glog.Fatalf("Backend url is invalid: %v", err)
+	}
+
+	backendProxy := httputil.NewSingleHostReverseProxy(backendURL)
+
+	// TODO: configure client certs on outbound connection to backend.
```
Comment (jonathanwei) [inline frontend_http.go]: Remove.

File: backend.go (hunk 1 above)
Comment (balasanjay) [inline backend.go]: Done.

**Pull request merged**